### PR TITLE
Migrate spin inbound-http test to integration test suite

### PR DESCRIPTION
**Pull request by lann**
Also sneak in a `use reqwest::Method`; it seemed weird that `reqwest` was appearing all over the test cases where its just an implementation detail.

### CODE REVIEW

File: tests/test-components/components/integration-spin-inbound-http/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return Err("request params field is deprecated".into());
+    }
+
+    let headers = req.headers.into_iter().collect::<HashMap<_, _>>();
+    let path = headers
+        .get("spin-path-info")
+        .map(|path| path.as_str())
+        .unwrap_or("");
+    match path {
+        "/echo" => Ok(req.body.unwrap_or_default()),
```
Comment (rylev) [inline tests/test-components/components/integration-spin-inbound-http/src/lib.rs:41]: One thing I'd like for us to try to do is to remove redundancy in our testing. We have other tests that test echoing capabilities as well as tests that read request headers. The unique elements of this test seem to be asserting that `params` is empty and reading `spin-path-info`. I wonder if this test can be simplified?

**Review by rylev — CHANGES_REQUESTED** [review]
I'd like to explore the other option that parts of this test can simply be deleted.

File: tests/test-components/components/integration-spin-inbound-http/src/lib.rs (hunk 1 above)
Comment (lann) [reply tests/test-components/components/integration-spin-inbound-http/src/lib.rs:41]: Don't the other tests only cover wasi handlers?

Comment (rylev) [reply tests/test-components/components/integration-spin-inbound-http/src/lib.rs:41]: Ah you're right. Ok - we should probably make tests that test the wasi handler and the Spin handler mirror one another so we can more easily see if there are gaps in test coverage, but that can be done in a follow up.

**Review by rylev — APPROVED** [review]

**Comment by rylev** [comment]
@lann you'll need to rebase for the CI fix

**Pull request merged**